File: recognition/src/main/java/br/com/alexandre/friday/recognize/chain/HelloChain.java
package br.com.alexandre.friday.recognize.chain;

import br.com.alexandre.friday.recognize.RecognizerChain;
import br.com.alexandre.friday.recognize.RecognizerRequest;

import java.util.Arrays;
import java.util.Collection;

public class HelloChain extends RecognizerChain {

    public Collection<String> execute(RecognizerRequest request) {
        return Arrays.asList(String.format("Olá, %s! Em que posso ser útil?", request.getName()));
    }

    public boolean apply(RecognizerRequest request) {
        return (request.getQ().contains("olá"));
    }
}

File: recognition/src/main/java/br/com/alexandre/friday/recognize/chain/HelloChain.kt
package br.com.alexandre.friday.recognize.chain

import br.com.alexandre.friday.recognize.RecognizerChain
import br.com.alexandre.friday.recognize.RecognizerRequest
import java.util.*

class HelloChain: RecognizerChain() {
    override fun execute(request: RecognizerRequest?): MutableCollection<String> {
        var response: String = if (request?.name == null) {
            "Olá! Em que posso ser útil?"
        } else {
            String.format("Olá, %s! Em que posso ser útil?", request.name);
        }
        return Arrays.asList(response)
    }

    override fun apply(request: RecognizerRequest?): Boolean {
        return request?.q!!.contains("olá")
    }
}